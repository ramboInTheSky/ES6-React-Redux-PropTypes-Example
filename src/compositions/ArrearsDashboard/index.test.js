import React from 'react';
import { shallow } from 'enzyme';

import { ArrearsDashboardComposition } from './';

describe('<ArrearsDashboardComposition />', () => {
  let props;
  let el;

  beforeEach(() => {
    props = {
      cardLabels: {
        address: 'Address',
        currentBalance: 'Current balance',
        heading: 'Arrears',
        notificationDate: 'Notification date',
        openTasks: 'Open tasks',
        tenantName: 'Tenant name',
      },
      taskCardLabels: {
        heading: 'Tasks',
        dueDate: 'Due Date',
        taskOwner: 'Task Owner',
        title: 'Title',
        customerName: 'Customer Name',
        address: 'Address',
        status: 'Status',
        overdue: 'overdue!',
      },
      arrears: [],
      tasks: [],
      arrearsNoDataMsg: 'Gotham is in danger',
      tasksNoDataMsg: 'Gotham is in danger',
      heading: 'titleText',
      updatePageHeader: jest.fn(),
      getTasksSummary: jest.fn(),
      getArrearStatuses: jest.fn(),
      getTaskStatuses: jest.fn(),
      arrearStatuses: {},
      taskStatuses: {},
      loadingArrears: false,
      loadingTasks: false,
      profile: {
        id: 'ABC123',
        patchName: 'Hello world',
      },
      patches: [
        {
          name: 'John Smith',
          patchName: 'Hello world',
        },
        {
          name: 'Jane Smith',
          patchName: 'Some other name',
        },
      ],
      getArrearsSummary: jest.fn(),
      onOpenPatchSelect: jest.fn(),
      invalidateTasks: jest.fn(),
    };

    el = shallow(<ArrearsDashboardComposition {...props} />);
  });

  it('should render the page', () => {
    expect(el).toMatchSnapshot();
  });

  it('should render the page while loading arrears and tasks', () => {
    el.setProps({
      loadingArrears: true,
      loadingTasks: true,
    }).update();
    expect(el).toMatchSnapshot();
  });

  it('should update the header', () => {
    expect(props.updatePageHeader).toHaveBeenCalled();
  });

  it('should call getTaskSummary', () => {
    expect(props.getTasksSummary).toHaveBeenCalled();
  });

  it('should call invalidateTasks on UnMount', () => {
    el.unmount();
    expect(props.invalidateTasks).toHaveBeenCalled();
  });
});
