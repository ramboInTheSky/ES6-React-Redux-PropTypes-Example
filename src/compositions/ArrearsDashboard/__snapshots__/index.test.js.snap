// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArrearsDashboardComposition /> should render the page 1`] = `
<withRouter(Connect(PageContentComposition))>
  <div
    className="col-lg-9"
  >
    <Connect(ArrearsFiltersComposition)
      onChange={[Function]}
    />
  </div>
  <div
    className="col-lg-9"
  >
    <ErrorBoundary
      messageOnScreen={null}
    >
      <WithTheme(Component)
        cardLabels={
          Object {
            "address": "Address",
            "customerName": "Customer Name",
            "dueDate": "Due Date",
            "heading": "Tasks",
            "overdue": "overdue!",
            "status": "Status",
            "taskOwner": "Task Owner",
            "title": "Title",
          }
        }
        loading={false}
        noDataMsg="Gotham is in danger"
        tasks={Array []}
        userId="ABC123"
        youText=""
      />
    </ErrorBoundary>
  </div>
  <aside
    className="col-lg-3"
  >
    <ErrorBoundary
      messageOnScreen={null}
    >
      <styled.div>
        <ActivePatchList
          isButtonBottomAligned={true}
          openPatchSelect={[Function]}
          patches={
            Array [
              Object {
                "customName": true,
                "fullname": "",
                "name": "",
                "patchName": "Hello world",
              },
              Object {
                "customName": false,
                "fullname": "Jane Smith",
                "name": "Jane Smith",
                "patchName": "Some other name",
              },
            ]
          }
        />
      </styled.div>
    </ErrorBoundary>
    <Connect(PatchStatisticsCompositions) />
  </aside>
</withRouter(Connect(PageContentComposition))>
`;

exports[`<ArrearsDashboardComposition /> should render the page while loading arrears and tasks 1`] = `
<withRouter(Connect(PageContentComposition))>
  <div
    className="col-lg-9"
  >
    <Connect(ArrearsFiltersComposition)
      onChange={[Function]}
    />
  </div>
  <div
    className="col-lg-9"
  >
    <Loader
      theme={
        Object {
          "colors": Object {
            "tertiaryLighter": "#D9D4D4",
          },
        }
      }
    />
  </div>
  <aside
    className="col-lg-3"
  >
    <ErrorBoundary
      messageOnScreen={null}
    >
      <styled.div>
        <ActivePatchList
          isButtonBottomAligned={true}
          openPatchSelect={[Function]}
          patches={
            Array [
              Object {
                "customName": true,
                "fullname": "",
                "name": "",
                "patchName": "Hello world",
              },
              Object {
                "customName": false,
                "fullname": "Jane Smith",
                "name": "Jane Smith",
                "patchName": "Some other name",
              },
            ]
          }
        />
      </styled.div>
    </ErrorBoundary>
    <Connect(PatchStatisticsCompositions) />
  </aside>
</withRouter(Connect(PageContentComposition))>
`;
